internal/appsec/dyngo/instrumentation/httpinstr/http.go

```diff
@@ -24,12 +26,7 @@ import (
 type (
 	// HandlerOperationArgs is the HTTP handler operation arguments.
 	HandlerOperationArgs struct {
-		Method     string
-		Host       string
-		RemoteAddr string
-		Path       string
-		IsTLS      bool
-		Span       ddtrace.Span
+		OnSecurityEvent func(event json.RawMessage)
```

Comment:
Looking for comments on this.

The problem I try to solve here is how can I report an attack from the WAF that will be handled in the integration package, so that the WAF code becomes "integration agnostic".

The idea is to pass a security event handler to the WAF and I chose here to simply pass it via the `HandlerOperationArgs`. It's kind of the explicit alternative of the storage of a value into a Go context that would be later retrieved. You can see the impact of this function callback in the WAF callback and the HTTP integration below: the span tags, that are HTTP-specific, are set in the HTTP instrumentation package `httpinstr`.

The downside is that the `HandlerOperationArgs` was supposed to only be a data input for the listeners and is suddenly used as a way to call an integration-specific handler function.

So I see two other probably cleaner alternatives:
1. Simple: use a Go context instead that would store this security event handler and pass it to the `dyngo` callbacks as first arguments. So `dyngo.StartOperation()` would become `dyngo.StartOperation(ctx context.Context, args interface{})`.
2. More complex: start appsec from the integration so that we can pass it a security event handler to use when it instantiates the WAF callback. So `appsec.registerWAF()` would become `appsec.registerWAF(rules []byte, handleEvent func (event json.RawMessage))`. It involves lots of changes in how appsec is started right now so that we could separately start it and register its security callbacks when an integration is used.